src/Core/Parser/Sequencer.php

```diff
@@ -122,6 +122,11 @@ class Sequencer
      */
     protected $nodeStack = [];
 
+    /**
+     * @var array<string, array<SequencingException>>
```

Comment:
`@var SequencingException[]` is preferred.

Reply:
`SequencingException[]` means `list<SequencingException>`, means `array<int, SequencingException>` - which does not reflect using `$nodeHash` as string.